woocommerce/woocommerce-ios
# Variations: Update product details screen after generating a variation

closes #3619 

# Why

[Creating a variation is already implemented](#3658), but screens like the product detail screen weren't notified of such creation, hence not updating its UI.

This PR makes sure that the `ProductFormViewController` is updated with the new variation after generating a new variation(and attributes)

# How

- Update `EditAttributesViewController` to provide an updated product(adding the new variation id) after generating a variation. Before, only the new variation was provided.

- Modify `AddAttributeViewController`, `EditAttributesViewController`, and `VariationsViewController` to provide closures for when the  product gets updated(with thew new variation and attributes) .

- Update `ProductFormViewController` to listen to the `VariationsViewController. onProductUpdate` closure

- Update `ProductFormViewModel` to correctly update the underlying product with the new variation, without losing any pending change that the merchant hasn't committed yet.

# Demo

https://user-images.githubusercontent.com/562080/108268770-04a71500-713b-11eb-87af-313adf460e56.mov


# Testing Steps

- Create a New Variable product
- Create a Variation by creating one or more attributes
- After generating the variation see that it renders properly in the variation list screen.
- When tapping back, see that the product details screen has information about the variation.

Update release notes:
